### PR TITLE
Feature/aesni backend

### PR DESCRIPTION
**Pull request by IceTDrinker**
### Resolves:

closes https://github.com/zama-ai/concrete-core-internal/issues/146
closes https://github.com/zama-ai/concrete-core-internal/issues/170
closes https://github.com/zama-ai/concrete-core-internal/issues/230

### Description

Implement the compile-time CSPRNG choice for DefaultEngine and DefaultParallelEngine

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

~~* [ ] Tests for the changes have been added (for bug fixes / features)~~
~~* [ ] Docs have been added / updated (for bug fixes / features)~~
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (apply the `aws_test` to the PR to launch the tests on AWS)
* [x] The draft release description has been updated
* [x] Check for breaking changes and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
tests running here: https://github.com/zama-ai/concrete-core/actions/runs/2501206706

I corrected something which would have broken non x86 targets

File: concrete-core/Cargo.toml [hunk 1]
```diff
@@ -33,9 +33,9 @@ default = ["backend_default", "seeder_unix"]
 doc = []
 
 # A pure-rust backend. Included by default in the build.
-backend_default = [
-    "concrete-csprng/generator_soft"
-]
+backend_default = ["concrete-csprng/generator_soft"]
+
+backend_default_generator_x86_64_aesni = ["concrete-csprng/generator_x86_64_aesni"]
```
Comment (aPere3) [inline concrete-core/Cargo.toml:38]: Maybe we could add a small comment explaining what this feature flag is doing exactly ?

**Review by aPere3 — APPROVED** [review]
Hello @IceTDrinker ,

Thanks for this ! It looks good for me ! 

I just have a minor comment regarding doc of feature flags (which we should definitely improve because we will have more and more of them).

Best,

Alex

File: concrete-core/Cargo.toml (hunk 1 above)
Comment (IceTDrinker) [reply concrete-core/Cargo.toml:40]: done

**Pull request merged**